lib/requestwrapper.ts

```diff
@@ -127,48 +127,30 @@ export class RequestWrapper {
 
     // Form params
     if (formData) {
-      // Remove keys with undefined/null values
-      // Remove empty objects
-      // Remove non-valid inputs for buildRequestFileObject,
-      // i.e things like {contentType: <contentType>}
       Object.keys(formData).forEach(key => {
-        if (formData[key] == null ||
-          isEmptyObject(formData[key]) ||
-          (formData[key].hasOwnProperty('contentType') && !formData[key].hasOwnProperty('data'))) {
-          delete formData[key];
-        }
-      });
-      // Convert file form parameters to request-style objects
-      Object.keys(formData).forEach(key => {
-        if (formData[key].data != null) {
-          formData[key] = buildRequestFileObject(formData[key]);
-        }
-      });
-
-      // Stringify arrays
-      Object.keys(formData).forEach(key => {
-        if (Array.isArray(formData[key])) {
-          formData[key] = formData[key].join(',');
-        }
-      });
-
-      // Convert non-file form parameters to strings
-      Object.keys(formData).forEach(key => {
-        if (!isFileParam(formData[key]) &&
-          !Array.isArray(formData[key]) &&
-          typeof formData[key] === 'object') {
-          (formData[key] = JSON.stringify(formData[key]));
-        }
-      });
-
-      // build multipart form data
-      Object.keys(formData).forEach(key => {
-        // handle files differently to maintain options
-        if (formData[key].value) {
-          multipartForm.append(key, formData[key].value, formData[key].options);
-        } else {
-          multipartForm.append(key, formData[key]);
-        }
+        const values = Array.isArray(formData[key]) ? formData[key] : [formData[key]];
+        // Skip keys with undefined/null values or empty object value
+        values.filter(v => v != null && !isEmptyObject(v)).forEach(value => {
```

Comment:
I think this will miss values of explicit type `undefined`. The method `isEmptyObject` is pretty strict about returning `true` iff the value passed in is the empty object `{}`. 

Something a little looser would catch more values:
```suggestion
        values.filter(v => v && !isEmptyObject(v)).forEach(value => {
```

Or we can just be more explicit:

`values.filter(v => v != null && v != undefined && !isEmptyObject(v)).forEach(value => {`

but I don't know if that's necessary.

Reply:
The filter as coded will eliminate "undefined" values.
```
[mkistler@mkistlers-mbp] ~>node
> foo = [ undefined ]
[ undefined ]
> foo.filter(v => v != null)
[]
```

Is there another case you are concerned about?

Reply:
Ah, my mistake. I was thinking about the results when using the more strict `!==` rather than `!=`. This looks good as is 👍